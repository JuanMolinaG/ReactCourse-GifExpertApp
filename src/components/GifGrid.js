import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const { data: gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__pulse'>{category}</h3>

      {loading && (
        <p className='animate__animated animate__fadeIn'>Loading...</p>
      )}

      <div className='card-grid'>
        {gifs.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
