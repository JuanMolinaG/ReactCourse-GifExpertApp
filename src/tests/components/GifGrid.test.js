import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {
  const category = 'Shiryu';

  test('Debe de mostrar el componente <GifGrid/> correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {
    const gifs = [
      {
        id: 'id',
        url: 'http://localhost/image.jpg',
        title: 'Hola Mundo',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
