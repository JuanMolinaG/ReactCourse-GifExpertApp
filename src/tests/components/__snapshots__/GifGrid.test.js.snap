// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe de mostrar el componente <GifGrid/> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    Shiryu
  </h3>
  <p
    className="animate__animated animate__fadeIn"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> Debe de mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__pulse"
  >
    Shiryu
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="id"
      key="id"
      title="Hola Mundo"
      url="http://localhost/image.jpg"
    />
  </div>
</Fragment>
`;
